src/toast/util/toastHooks.ts

```diff
@@ -0,0 +1,77 @@
+import {useCallback, useContext} from "react";
+
+import {generateRandomString} from "../../core/utils/string/stringUtils";
+import {ToastContext} from "../ToastProvider";
+import {ToastData} from "./toastTypes";
+
+/**
+ * @returns {Object} Current value of ToastContext
+ */
+function useToastContext() {
+  return useContext(ToastContext);
```

Comment:
Let's throw an error if the context is being used outside of a Provider component tree